schedulingApp/views.py

```diff
@@ -183,16 +186,28 @@ def post(self, request):
                                                        "tas": Profile.objects.filter(permission=Profile.TA),
                                                        "courses": Course.objects.all()})
 
-
-@method_decorator(user_passes_test(user_has_admin_permission), name='dispatch')
+@method_decorator(user_passes_test(user_has_professor_permission), name='dispatch')
 class AssignSectionUser(View):
     def get(self, request, id):
         section = Section.objects.get(id=id)
-        return render(request, "assignSectionUser.html", {"profile": Profile.objects.get(user=request.user),
-                                                          "course": section,
-                                                          "courseUsers": section.getUsersAssignedToCourse(),
-                                                          "users": Profile.objects.filter(Q(permission=Profile.TA) | Q(
-                                                              permission=Profile.PROFESSOR))})
+        profile = Profile.objects.get(user=request.user)
+        courseAssignedUsers = section.course.getUsersAssignedToCourse()
+        if profile.permission == Profile.ADMIN:
+            return render(request, "assignSectionUser.html", {"profile": Profile.objects.get(user=request.user),
+                                                              "course": section,
+                                                              "courseUsers": section.getUsersAssignedToSection(),
+                                                              "users": Profile.objects.filter(
```

Comment:
make users a variable, don't repeat yourself

Reply:
Refactored and committed.